src/CitizenInitiative/ActivitySubscriptionManager.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace AppBundle\CitizenInitiative;
+
+use AppBundle\Entity\ActivitySubscription;
+use AppBundle\Entity\Adherent;
+use AppBundle\Repository\ActivitySubscriptionRepository;
+use Doctrine\Common\Persistence\ObjectManager;
+
+class ActivitySubscriptionManager
+{
+    private $repository;
+    private $manager;
+
+    public function __construct(
+        ActivitySubscriptionRepository $repository,
+        ObjectManager $manager
+    ) {
+        $this->repository = $repository;
+        $this->manager = $manager;
+    }
+
+    public function subscribeToAdherentActivity(Adherent $following, Adherent $followed): void
+    {
+        $activitySubscription = $this->repository->findSubscription($following, $followed);
+
+        if (!$activitySubscription) {
+            $activitySubscription = new ActivitySubscription($following, $followed);
+        } elseif ($activitySubscription->isSubscribed()) {
```

Comment:
It would have been better to have more domain specific methods like

- `subscribe()`
- `unsubscribe()`
- `toggleSubscription()`